### DISCUSSION TITLE
Mass Actions in Grid

### DISCUSSION THREAD

**Discussion by mucan54** [orchidsoftware/platform#1468] [Ideas]
Maybe we can make mass actions in grid with checkbox at the first column and maybe a select box at the top of grid to select witch action will be done with selected rows.

**Comment by blankster** [comment]
+1 for this idea!

**Comment by tabuna** [comment]
Hey @mucan54. We can already specify the checkbox in the table, for example:

```php
TD::set('id')
    ->render(function (User $user){
        return CheckBox::make('users[]')
            ->value($user->id)
            ->placeholder($user->getNameTitle())
            ->checked(false);
    });
```

With any method of processing, the values will be sent and can be bulk processed. (#1468)
The only thing is that there is no beautiful layer for placing action selection buttons.


![image](https://user-images.githubusercontent.com/5102591/102691670-2bba9680-421f-11eb-88bb-5ba1cb62d499.png)


I thought of copying this concept for the CRUD package since the rules are much stricter than for general screens. 
What do you think about implementing this specifically for https://github.com/orchidsoftware/crud? 


But so far, I was in no hurry with this since I never had a real need for mass processing of records that need to be selected manually. Would you like to work on this implementation?

**Reply by mucan54** [reply]
Sure, I can work for it.

**Comment by vallter2** [comment]
Any chance to add Listener to Switcher component in table layout? When change true/false to call function?

**Reply by mahdi-above** [reply]
I am searching for this too.

**Comment by moskoweb** [comment]
Hello, how do you use bulk actions? Why did vi to put the checkbox, but how to make the button at the top grab the selected checkbox?

**Comment by Davka** [comment]
It would be possible to add a row at the end of the table containing a select field and a button. I have already used the total row elsewhere.
<img width="1187" alt="Bildschirmfoto 2024-02-26 um 11 35 27" src="https://github.com/orchidsoftware/platform/assets/1117697/a232d8a3-83f3-42d1-b9f2-6c52b0c0e222">

**Comment by naquad** [comment]
I have a case where exactly batch actions are required. I've tried to implement something but drowned in a complete absence of integration between JS controllers (remembering the selected items + pagination + recovering the selection state + a button that appears only when something is selected).

I suspect I lack some knowledge about the JS side, but before going too deep, is that doable at the moment?
